File: c++/14mod3.cpp
#include <iostream>
using namespace std;

bool isPrime(int n){ // Функция для проверки простого числа
    if (n < 2) return false;
    for (int i = 2; i * i <= n; i++){ // Цикл идет до sqrt(n) 
        if (n % i == 0) return false;
    }
    return true;
}


int main(){
    int n, count = 0;
    cout << "Введите количество чисел" << endl;
    cin >> n;
    for (int i = 0; i < n; i++){
        int num;
        cout << i + 1 << " Число: ";
        cin >> num;
        if (isPrime(num)) count++;
    }
    cout << count << " Простых чисел(а)" << endl;
    return 0;
}

File: python/14mod3.py
import math

def isPrime(n):
    if n < 2:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):  # проверка до квадратного корня числа
        if n % i == 0:
            return False
    return True

def main():
    size = int(input("Введите количество чисел "))
    count = 0
    for i in range(size):
        num = int(input(f"{i + 1} Число = "))
        if isPrime(num) == True:
            count += 1
    print(f"Простых чисел {count}")

if __name__ == "__main__":
    main()

